src/agent/src/mount.rs

```diff
@@ -50,7 +50,76 @@ const RW_MASK: u32 = 0o660;
 const RO_MASK: u32 = 0o440;
 const EXEC_MASK: u32 = 0o110;
 const MODE_SETGID: u32 = 0o2000;
+pub trait RefCounted {
```

Comment:
Do you intend to introduce other types that implement this trait?

If not, why are you defining it instead of just implementing these functions directly for the `RefCountedObject` type?

Reply:
changed